modules/gathering-clusterversion-history.adoc

```diff
@@ -0,0 +1,70 @@
+// Module included in the following assemblies:
+//
+// * updating/troubleshooting_updates/gathering-data-cluster-update.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="gathering-clusterversion-history_{context}"]
+= Gathering ClusterVersion history 
+
+When updating your cluster, the cluster version operator (CVO) creates a `status.history` entry to record the update. To gather and understand the history of your successful or failed cluster update, it is important to retrieve the ClusterVersion history.
+
+[NOTE]
+====
+The ClusterVersion history has a size limit, so in cases where enough updates occurred to have reached that limit, the least relevant entries in the history are pruned to accommodate the limit. 
+====
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` role.
+* You have installed the OpenShift CLI (`oc`).
+
+.Procedure
+
+. View a detailed summary of cluster specifications, update availability, and update history:
++
+[source,terminal]
+----
+$ oc get clusterversion version -o json | jq .status.history
```

Comment:
Dropping the `jq` exposure (you don't list it as a prereq, and reducing the number of prereqs is good) with [JSONPath][1]:

```console
$ oc get -o 'jsonpath={range .status.history[*]}{@}{"\n"}{end}' clusterversion version
{"completionTime":"2023-11-07T20:26:04Z","image":"quay.io/openshift-release-dev/ocp-release@sha256:a148b19231e4634196717c3597001b7d0af91bf3a887c03c444f59d9582864f4","startedTime":"2023-11-07T19:11:36Z","state":"Completed","verified":true,"version":"4.13.19"}
{"completionTime":"2023-10-04T18:53:29Z","image":"quay.io/openshift-release-dev/ocp-release@sha256:eac141144d2ecd6cf27d24efe9209358ba516da22becc5f0abc199d25a9cfcec","startedTime":"2023-10-04T17:26:31Z","state":"Completed","verified":true,"version":"4.13.13"}
{"completionTime":"2023-09-26T14:21:43Z","image":"quay.io/openshift-release-dev/ocp-release@sha256:371328736411972e9640a9b24a07be0af16880863e1c1ab8b013f9984b4ef727","startedTime":"2023-09-26T14:02:33Z","state":"Completed","verified":false,"version":"4.13.12"
```

That's not quite as pretty as the newlined output that `jq` produces, but it's not completely illegible.  Another alternative with more opinions about how to format the properties in each entry:

```console
$ oc get -o 'jsonpath={range .status.history[*]}{.startedTime}{" "}{.completionTime}{" "}{.state}{" "}{.version}{" (verified: "}{.verified}{", "}{.image}{") "}{.acceptedRisks}{"\n"}{end}' clusterversion version
2023-11-07T19:11:36Z 2023-11-07T20:26:04Z Completed 4.13.19 (verified: true, quay.io/openshift-release-dev/ocp-release@sha256:a148b19231e4634196717c3597001b7d0af91bf3a887c03c444f59d9582864f4) 
2023-10-04T17:26:31Z 2023-10-04T18:53:29Z Completed 4.13.13 (verified: true, quay.io/openshift-release-dev/ocp-release@sha256:eac141144d2ecd6cf27d24efe9209358ba516da22becc5f0abc199d25a9cfcec) 
2023-09-26T14:02:33Z 2023-09-26T14:21:43Z Completed 4.13.12 (verified: false, quay.io/openshift-release-dev/ocp-release@sha256:371328736411972e9640a9b24a07be0af16880863e1c1ab8b013f9984b4ef727)
```

There's also:

```console
$ oc describe clusterversions/version
...
  Desired:
    Channels:
      candidate-4.13
      candidate-4.14
      fast-4.13
      fast-4.14
      stable-4.13
    Image:    quay.io/openshift-release-dev/ocp-release@sha256:a148b19231e4634196717c3597001b7d0af91bf3a887c03c444f59d9582864f4
    URL:      https://access.redhat.com/errata/RHSA-2023:6130
    Version:  4.13.19
  History:
    Completion Time:    2023-11-07T20:26:04Z
    Image:              quay.io/openshift-release-dev/ocp-release@sha256:a148b19231e4634196717c3597001b7d0af91bf3a887c03c444f59d9582864f4
    Started Time:       2023-11-07T19:11:36Z
    State:              Completed
    Verified:           true
    Version:            4.13.19
    Completion Time:    2023-10-04T18:53:29Z
    Image:              quay.io/openshift-release-dev/ocp-release@sha256:eac141144d2ecd6cf27d24efe9209358ba516da22becc5f0abc199d25a9cfcec
    Started Time:       2023-10-04T17:26:31Z
    State:              Completed
    Verified:           true
    Version:            4.13.13
    Completion Time:    2023-09-26T14:21:43Z
    Image:              quay.io/openshift-release-dev/ocp-release@sha256:371328736411972e9640a9b24a07be0af16880863e1c1ab8b013f9984b4ef727
    Started Time:       2023-09-26T14:02:33Z
    State:              Completed
    Verified:           false
    Version:            4.13.12
  Observed Generation:  4
  Version Hash:         CMLl3sLq-EA=
Events:                 <none>
```

if we think that pretty-formatting of the `history` part is worth the hassle of scrolling through the non-history part of the structure.

And if we expect this to be something that folks need with regularity, in isolation from other parts of ClusterVersion, we can add new `oc ...` subcommands to make it easy to get at this information without needing `jq` or JSONPath.

[1]: https://kubernetes.io/docs/reference/kubectl/jsonpath/

Reply:
Also, these options are not exclusive.  You could present both a JSONPath example and a `jq` example if you like.